### PR TITLE
HDDS-7765. [Snapshot] Handle OzoneManager#getKeyInfo with OmMetadataReader.

### PR DESCRIPTION
**Pull request by GeorgeJahad**
## What changes were proposed in this pull request?

Merge from master was incomplete in that it did not move the new getKeyInfo() method from OzoneManager to the OmMetadataReader,( and OmMetadataReaderMetrics).

This PR rectifies that.

It also modifies the OmMetadataManagerImpl to not use the unneeded tableCacheMetrics on snapshots.  (These metrics were added by the merge and are not used by snapshots.)

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-7765

## How was this patch tested?

Tested against the acceptance test for this PR: https://github.com/apache/ozone/pull/4134

### CODE REVIEW

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshot.java [hunk 1]
```diff
@@ -215,6 +225,12 @@ private  OzoneFileStatus denormalizeOzoneFileStatus(
         omKeyInfo, fileStatus.getBlockSize(), fileStatus.isDirectory());
   }
 
+  private KeyInfoWithVolumeContext denormalizeKeyInfoWithVolumeContext(
+      KeyInfoWithVolumeContext k) {
+    return new KeyInfoWithVolumeContext(k.getVolumeArgs().orElse(null),
+        k.getUserPrincipal().orElse(null),
+        denormalizeOmKeyInfo(k.getKeyInfo()));
+  }
```
Comment (smengcl) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshot.java:233]: nit
```suggestion
  }

```

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshot.java [hunk 2]
```diff
@@ -215,6 +225,12 @@ private  OzoneFileStatus denormalizeOzoneFileStatus(
         omKeyInfo, fileStatus.getBlockSize(), fileStatus.isDirectory());
   }
 
+  private KeyInfoWithVolumeContext denormalizeKeyInfoWithVolumeContext(
+      KeyInfoWithVolumeContext k) {
+    return new KeyInfoWithVolumeContext(k.getVolumeArgs().orElse(null),
```
Comment (smengcl) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshot.java:230]: Would using its `Builder` be better than calling the constructor directly?

**Review by smengcl — COMMENTED** [review]
Overall look good to me

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshot.java (hunk 2 above)
Comment (GeorgeJahad) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshot.java]: done

**Review by smengcl — APPROVED** [review]
+1, pending CI.

**Comment by smengcl** [comment]
Thanks @GeorgeJahad for taking care of this.

@xBis7 @aswinshakil @swamirishi You may rebased your patches on to the latest snapshot feature to see if that solves the (hanging) issues.

**Comment by GeorgeJahad** [comment]
FYI: it did solve the hanging problem for me, so it should solve the others unless they are significantly different.  Thanks for the review @smengcl !

**Pull request merged**